libsolidity/formal/CHC.cpp

```diff
@@ -525,6 +527,39 @@ void CHC::visitAssert(FunctionCall const& _funCall)
 	m_context.addAssertion(m_error.currentValue() == previousError);
 }
 
+void CHC::internalFunctionCall(FunctionCall const& _funCall)
+{
+	solAssert(m_currentContract, "");
+
+	auto const* function = functionCallToDefinition(_funCall);
+	if (function)
+	{
+		if (m_currentFunction && !m_currentFunction->isConstructor())
+			m_callGraph[m_currentFunction].insert(function);
+		else
+			m_callGraph[m_currentContract].insert(function);
+		auto const* contract = function->annotation().contract;
+
+		// Liraries can have constants as their "state" variables,
+		// so we need to ensure they were constructed correctly.
+		if (contract->isLibrary())
+			m_context.addAssertion(interface(*contract));
+	}
+
+	auto previousError = m_error.currentValue();
+
+	m_context.addAssertion(predicate(_funCall));
+
+	connectBlocks(
+		m_currentBlock,
+		(m_currentFunction && !m_currentFunction->isConstructor()) ? summary(*m_currentFunction) : summary(*m_currentContract),
+		(m_error.currentValue() > 0)
+	);
+	m_context.addAssertion(m_error.currentValue() == 0);
+	m_error.increaseIndex();
+	m_context.addAssertion(m_error.currentValue() == previousError);
```

Comment:
The whole idea here is:
- Jump to the summary of the current function (or contract in case of constructor) if the error flag returned by the called function is non-zero
- Otherwise assert that the returned error flag is 0, and continue to the next block with the error flag from before
I don't think there are any circumstances where `previousError` is not 0 there, but wanted to keep the model consistent.

Reply:
Alright, then I got that correctly - not sure why to keep this assertion then, but there's also no harm in it, so fine with me!

Reply:
Just wanted to be on the safe side. If a bug is introduced in the error flag encoding at some point, this could maybe mask it, or something like that